paginator/paginator.go

```diff
@@ -221,13 +227,19 @@ func (p *Paginator) appendPagingQuery(db *gorm.DB, fields []interface{}) *gorm.D
 	stmt := db
 	stmt = stmt.Limit(p.limit + 1)
 	stmt = stmt.Order(p.buildOrderSQL())
-	if len(fields) > 0 {
-		stmt = stmt.Where(
-			p.buildCursorSQLQuery(),
-			p.buildCursorSQLQueryArgs(fields)...,
-		)
+
+	if len(fields) == 0 {
+		return stmt
 	}
-	return stmt
+
+	if p.allowTupleCmp && p.canOptimizePagingQuery() {
+		return stmt.Where(p.buildOptimizedCursorSQLQuery(), fields...)
```

Comment:
I think that `buildOptimizedCursorSQLQuery` could just return a single `?` in the query string and then we could pass the entire `fields` array into `Where` without unpacking it with `...`.

Reply:
Updated.